File: models/employee.js
const mongoose = require('mongoose');

const employeeSchema = new mongoose.Schema({
    // البيانات الشخصية الأساسية
    name: { 
        type: String, 
        required: [true, 'اسم الموظف مطلوب'], 
        trim: true 
    },
    nationalId: { 
        type: String, 
        required: [true, 'رقم البطاقة مطلوب'],
        unique: true,
        trim: true 
    },
    email: { 
        type: String, 
        lowercase: true, 
        trim: true 
    },
    phone: { type: String, trim: true },

    // البيانات الوظيفية (Core HR)
    companyId: { 
        type: mongoose.Schema.Types.ObjectId, 
        ref: 'Company', 
        required: true 
    },
    department: { 
        type: mongoose.Schema.Types.ObjectId, 
        ref: 'Department', 
        required: [true, 'يجب تحديد قسم للموظف'] 
    },
    jobTitle: { type: String, required: true, trim: true },
    
    // نظام المدير المباشر (Org Chart Logic)
    reportingTo: { 
        type: mongoose.Schema.Types.ObjectId, 
        ref: 'Employee', 
        default: null 
    },

    // تواريخ التعيين والاستقالة
    hireDate: { 
        type: Date, 
        required: [true, 'تاريخ التعيين مطلوب'] 
    },
    resignationDate: { 
        type: Date, 
        default: null 
    },
    employmentType: { 
        type: String, 
        enum: ['Full-time', 'Part-time', 'Contractor'], 
        default: 'Full-time' 
    },

    // هيكل الرواتب المطور (Payroll Engine)
    salaryDetails: {
        basicSalary: { type: Number, required: true, default: 0 },
        // تم تحديث البدلات والحوافز لتكون مرنة مع الـ Logic الجديد
        additions: [{
            name: { type: String }, // اسم البند (مثلاً: حافز إنتاج)
            amount: { type: Number, default: 0 },
            type: { 
                type: String, 
                enum: ['Exempted', 'Non-Exempted'], // التعديل المطلوب لـ Taxable/Non-Taxable
                default: 'Non-Exempted' 
            }
        }],
        deductions: [{
            name: { type: String }, // اسم الخصم (مثلاً: تأمين طبي)
            amount: { type: Number, default: 0 },
            isMedical: { type: Boolean, default: false }, // علامة عشان الـ 15% Medical Rule
            type: { 
                type: String, 
                enum: ['Exempted', 'Non-Exempted'], 
                default: 'Non-Exempted' 
            }
        }]
    },

    // إدارة الوثائق (Document Management)
    documents: [{
        title: { type: String }, 
        fileUrl: { type: String }, 
        expiryDate: { type: Date }
    }],

    status: { 
        type: String, 
        enum: ['Active', 'Inactive', 'Resigned'], 
        default: 'Active' 
    },
    createdAt: { type: Date, default: Date.now }
});

// Indexing لسرعة البحث في الشركة والقسم
employeeSchema.index({ companyId: 1, department: 1 });
employeeSchema.index({ nationalId: 1 }, { unique: true });

/**
 * التصدير النهائي (Vercel Optimized)
 * يمنع الـ Duplicate model name error تماماً
 */
module.exports = mongoose.models.Employee || mongoose.model('Employee', employeeSchema);

File: models/employee.go
package models

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

// Employee يمثل هيكل بيانات الموظف في النظام
type Employee struct {
	// التعديل: خليناه _id في الـ json عشان يربط مع الـ Frontend والـ MongoDB صح
	ID              primitive.ObjectID  `bson:"_id,omitempty" json:"_id"` 
	Name            string              `bson:"name" json:"name"`
	NationalID      string              `bson:"nationalId" json:"nationalId"`
	Email           string              `bson:"email" json:"email"`
	Phone           string              `bson:"phone" json:"phone"`
	CompanyID       primitive.ObjectID  `bson:"companyId" json:"companyId"`
	Department      string              `bson:"department" json:"department"`
	JobTitle        string              `bson:"jobTitle" json:"jobTitle"`
	
	ReportingTo     *primitive.ObjectID `bson:"reportingTo,omitempty" json:"reportingTo"` 
	
	// التعديل: غيرنا النوع لـ string عشان يستقبل تاريخ الـ HTML (YYYY-MM-DD) من غير مشاكل Parsing
	HireDate        string              `bson:"hireDate" json:"hireDate"`
	ResignationDate *time.Time          `bson:"resignationDate,omitempty" json:"resignationDate"`
	
	EmploymentType  string              `bson:"employmentType" json:"employmentType"` 
	SalaryDetails   SalaryDetails       `bson:"salaryDetails" json:"salaryDetails"`
	
	History         []HistoryRecord     `bson:"history,omitempty" json:"history"`
	Documents       []Document          `bson:"documents,omitempty" json:"documents"`
	
	Status          string              `bson:"status" json:"status"` 
	CreatedAt       time.Time           `bson:"createdAt" json:"createdAt"`
}

// SalaryDetails تفاصيل الراتب والبدلات والخصومات
type SalaryDetails struct {
	BasicSalary       float64     `bson:"basicSalary" json:"basicSalary"`
	Transportation    float64     `bson:"transportation" json:"transportation"` 
	NetToGrossTarget  float64     `bson:"netToGrossTarget,omitempty" json:"netToGrossTarget"` 
	Additions         []Component `bson:"additions,omitempty" json:"additions"`
	Deductions        []Component `bson:"deductions,omitempty" json:"deductions"`
}

// Component يمثل بند مالي ديناميكي
type Component struct {
	Name       string  `bson:"name" json:"name"`
	Amount     float64 `bson:"amount" json:"amount"`
	Type       string  `bson:"type" json:"type"` 
	IsMedical  bool    `bson:"isMedical" json:"isMedical"`
}

// HistoryRecord سجل المرتبات الشهري
type HistoryRecord struct {
	Month   string      `bson:"month" json:"month"`
	Payload interface{} `bson:"payload" json:"payload"` 
}

// Document الوثائق المرفقة للموظف
type Document struct {
	Title      string     `bson:"title" json:"title"`
	FileURL    string     `bson:"fileUrl" json:"fileUrl"`
	ExpiryDate *time.Time `bson:"expiryDate,omitempty" json:"expiryDate"`
}